render/egl.c

```diff
@@ -358,6 +358,58 @@ struct wlr_egl *wlr_egl_create(EGLenum platform, void *remote_display,
 	return NULL;
 }
 
+struct wlr_egl *wlr_egl_create_from_drm_fd(int drm_fd) {
+	PFNEGLQUERYDEVICESTRINGEXTPROC eglQueryDeviceStringEXT;
+	PFNEGLQUERYDEVICESEXTPROC eglQueryDevicesEXT;
+
+	load_egl_proc(&eglQueryDevicesEXT, "eglQueryDevicesEXT");
+	load_egl_proc(&eglQueryDeviceStringEXT,"eglQueryDeviceStringEXT");
+
+	static const EGLint MAX_DEVICES = 64;
+	EGLint nb_devices = 0;
+
+	eglQueryDevicesEXT(MAX_DEVICES, NULL, &nb_devices);
```

Comment:
I'd prefer to enable EGL debugging prior to any EGL calls (see `eglDebugMessageControlKHR` in `wlr_egl_create`).

Reply:
I ended up splitting the `wlr_egl_create` function in two to get the basic EGL functions first, and then initialize the rest.